addons/event_sale/wizard/event_configurator.py

```diff
@@ -11,3 +12,35 @@ class EventConfigurator(models.TransientModel):
     product_id = fields.Many2one('product.product', string="Product", readonly=True)
     event_id = fields.Many2one('event.event', string="Event")
     event_ticket_id = fields.Many2one('event.event.ticket', string="Event Ticket")
+    has_available_events = fields.Boolean(
+        "Has Available Events",
+        compute="_compute_has_available_events",
+        help="Technical field used to know if the user has at least one event that can be selected and used to configure event tickets on sale order lines")
+
+    @api.model
+    def default_get(self, fields):
+        res = super(EventConfigurator, self).default_get(fields)
+        if 'event_id' in fields and res.get('product_id'):
+            upcoming_events = self.env['event.event'].search(
+                [('event_ticket_ids.product_id', '=', res['product_id']),
+                 ('date_end', '>=',
+                  date.today().strftime('%Y-%m-%d 00:00:00')),
+                 ('stage_id.pipe_end', '=', False), ], limit=2)
+            if 'event_ticket_id' in fields and len(upcoming_events) == 1:
```

Comment:
Mmmh almost ^^

I think the `if 'event_ticket_id'` check can be done on the `if len(upcoming_events.event_ticket_ids) == 1` instead, it makes more sense there, don't you think?

Reply:
Oh, yeah as this could lead to errors actually, it has now been corrected.